### PR TITLE
feat: support prerelease version increments

### PR DESCRIPTION
**Pull request by Andrapyre**
This PR adds support for incrementing prerelease versions by default, if it ends in a number. Currently, if a prerelease version is incremented, the prerelease qualifier is simply dropped. E.g. `1.0.0-RC1` will be incremented to `1.0.0`. After this merge, `1.0.0-RC1` will be incremented to `1.0.0-RC2`, but prerelease versions without a version number will behave as before: `1.0.0-alpha` will be incremented to `1.0.0`.

Additionally, the `.asSnapshot` method will include the prerelease version. E.g. `1.0.0-RC1` will return `1.0.0-RC1-SNAPSHOT`. 

This PR also adds type declarations for public members in the `Version` class, as well as some ScalaDocs.

### CODE REVIEW

**Comment by Andrapyre** [comment]
@xuwei-k , is this project still active? It looks like it's been two years since the last release. Not sure if you're open to taking on any official maintainers, but happy to help in that way or just with regular contributions (building for Scala 3, etc.). This plugin has been super helpful in my work. Thanks for keeping it going!

File: build.sbt [hunk 1]
```diff
@@ -26,25 +31,11 @@ Compile / doc / scalacOptions ++= {
...
-  }</developers>
-}
+ThisBuild / developers := List(
+  Developer(
+      id = "xuwei-k",
+      name = "Kenji Yoshida",
+      email = "@xuwei-k",
+      url = url("https://github.com/xuwei-k")
+  )
+)
```
Comment (xuwei-k) [inline build.sbt:41]: Please revert some unnecessary changes.

**Comment by Andrapyre** [comment]
@xuwei-k , I've removed all changes to `build.sbt` except standardizing the scala version. I also realized that `.withoutQualifier` is no longer being used in the code base after my changes, so I've marked it as deprecated in favor of `.withoutSnapshot`.

File: build.sbt [hunk 2]
```diff
@@ -5,7 +5,7 @@ name := "sbt-release"
 
 homepage := Some(url("https://github.com/sbt/sbt-release"))
 licenses := Seq("Apache-2.0" -> url("http://www.apache.org/licenses/LICENSE-2.0"))
-
+scalaVersion := "2.12.18"
```
Comment (xuwei-k) [inline build.sbt:8]:
```suggestion

```

**Comment by Andrapyre** [comment]
@xuwei-k , I've reverted the scala version change, as you've requested, but just curious for some context - isn't it best practice to declare scala versions in the build config so as to avoid different local environments building with a different scala version?

**Comment by Andrapyre** [comment]
@xuwei-k , now that the changes to build.sbt are reverted, would you be able to merge this PR?

**Comment by SethTisue** [comment]
> isn't it best practice to declare scala versions in the build config so as to avoid different local environments building with a different scala version?

sbt plugins are a bit unusual in this respect. The Scala version used is determined by the sbt version being used, so it's not really necessary to also pin the Scala version by other means.
